stko/molecular/topology_extractor/topology_extractor.py

```diff
@@ -0,0 +1,73 @@
+"""
+Topology Extractor
+==================
+
+Class for defining a topology from a molecule and disconnections.
+
+"""
+
+import logging
+
+from ..networkx import Network
+
+logger = logging.getLogger(__name__)
+
+
+class TopologyExtractor:
+    """
+    Extractor of topology definitions from a molecule.
+
+    """
+
+    def extract_topology(
+        self,
+        molecule,
+        atom_ids_to_disconnect,
+    ):
+
+        flat_tuple = []
+        for i in atom_ids_to_disconnect:
+            flat_tuple.append(i[0])
+            flat_tuple.append(i[1])
+        flat_tuple = tuple(set(flat_tuple))
```

Comment:
I would not make `flat_tuple` a `tuple`. You are doing a fair amount of membership lookups later (`if aid in flat_tuple`) -> these are at a cost of O(N)  for a tuple but O(1) for a set.

Reply:
Agreed